### ISSUE TITLE
[Good First Issue]: Align behavior of ONNX Frontend function ReduceL2-11, 13, 18 with original framework

### ISSUE THREAD

**Issue by gkrivor** [openvinotoolkit/openvino#20560]
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.
This task requires alignment between OpenVINO ONNX Frontend and original framework implementations of ReduceL2 for next list of opsets: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceL2-11),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceL2-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceL2-18)
Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

First of all, please, take a look on [ReduceMax](https://github.com/openvinotoolkit/openvino/pull/23475) PR for a reference.

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#ReduceL2)
More details can be found in ONNX Changelog: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceL2-11),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceL2-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceL2-18)
1. Function already has a common implementation in OpenVINO. First of all, you need to review a documentation and prepare a table with differences between versions. It could be, for instance, a missing property, extended/reduced coverage of existing property, etc...
2. Copy existing implementation [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op) to make it aligned with original framework (extend or reduce coverage of a common implementation). Copy of modified implementation should be in a defined opset, or in opset 1 in case it implements oldest implementation. [Example of multi-opset operation](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/softmax.cpp#L50).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Intel DevHub Discord channel](https://discord.gg/wPuqAujS) - engage in discussions, ask questions and talk to OpenVINO developers
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [User documentation](https://docs.openvino.ai/)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

### Contact points

@gkrivor

**Comment by LucaTamSapienza** [comment]
.take

**Comment by LucaTamSapienza** [comment]
|Version|Changes|
|:-------:|:-------:|
|ReduceL2-1|initial version|
|ReduceL2-11|`axes` range is now [-r, -r-1], r = rank(data)|
|ReduceL2-13|added support for `T`:  tensor(bfloat16)|
|ReduceL2-18|`axes` attributes is now `noop_with_empty_axes` : int (default is 0) <br> now the inputs can be non-differentiable `axes` (optional, non-differentiable) : tensor(int64)|

**Comment by LucaTamSapienza** [comment]
Hi, currently I'm adding namespaces set_11, 13, 18 inside reduce.cpp and reduce.hpp. I have registered the functions inside ops_bridge.cpp. Before creating the tests, I would like to have some feedback to know if I am proceeding in the right way. I have some questions about how to implement the changes that I've mentioned above.

**Comment by mlukasze** [comment]
you can prepare draft PR to show us changes you've prepared - it may make discussion easier :)

**Comment by LucaTamSapienza** [comment]
I apologize for the delay; I have some work to do. Here is my draft PR #22741

**Comment by mlukasze** [comment]
No worries, thanks for the draft.
